### PR TITLE
Set up test pipeline with jld2 restart for examples/3dsphere/

### PR DESCRIPTION
**Pull request by jiahe23**
This PR adds the following functionalities
-  output prognostic state to jld2 file at the user specified frequency,
-  restart the simulation by loading a jld2 state using an environment variable.

closes #541

### CODE REVIEW

**Comment by jiahe23** [comment]
bors try

**Review by bischtob — APPROVED** [review]
Looks good to me.

File: examples/3dsphere/balanced_flow_rhoe.jl [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
     zmax = FT(30.0e3),
     velem = 10,
     helem = 4,
     npoly = 4,
-    tmax = FT(60 * 60),
-    dt = FT(5.0),
-    ode_algorithm = OrdinaryDiffEq.SSPRK33,
+    tmax = FT(60 * 60 * 24 * 10),
+    dt = FT(400.0),
+    ode_algorithm = OrdinaryDiffEq.Rosenbrock23,
```
Comment (jiahe23) [inline examples/3dsphere/balanced_flow_rhoe.jl:17]:
```suggestion
    tmax = FT(60 * 60),
    dt = FT(5.0),
    ode_algorithm = OrdinaryDiffEq.SSPRK33,
```

File: examples/3dsphere/balanced_flow_rhoe.jl [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
     jacobian_flags = (; ∂𝔼ₜ∂𝕄_mode = :no_∂P∂K, ∂𝕄ₜ∂ρ_mode = :exact),
     max_newton_iters = 2,
-    save_every_n_steps = 10,
-    additional_solver_kwargs = (;), # e.g., reltol, abstol, etc.
+    save_every_n_steps = 2160,
```
Comment (jiahe23) [inline examples/3dsphere/balanced_flow_rhoe.jl:20]:
```suggestion
    save_every_n_steps = 10,
```

File: examples/3dsphere/balanced_flow_rhotheta.jl [hunk 3]
```diff
@@ -13,8 +16,8 @@ driver_values(FT) = (;
     ode_algorithm = OrdinaryDiffEq.Rosenbrock23,
     jacobian_flags = (; ∂𝔼ₜ∂𝕄_mode = :exact, ∂𝕄ₜ∂ρ_mode = :exact),
     max_newton_iters = 2,
-    save_every_n_steps = 216,
-    additional_solver_kwargs = (;),
+    save_every_n_steps = 2160,
```
Comment (jiahe23) [inline examples/3dsphere/balanced_flow_rhotheta.jl:19]:
```suggestion
    save_every_n_steps = 216,
```

File: examples/3dsphere/baroclinic_wave_rhoe.jl [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
     npoly = 4,
     tmax = FT(60 * 60 * 24 * 10),
-    dt = FT(500.0),
+    dt = FT(400.0),
     ode_algorithm = OrdinaryDiffEq.Rosenbrock23,
     jacobian_flags = (; ∂𝔼ₜ∂𝕄_mode = :no_∂P∂K, ∂𝕄ₜ∂ρ_mode = :exact),
     max_newton_iters = 2,
-    save_every_n_steps = 10,
-    additional_solver_kwargs = (;), # e.g., reltol, abstol, etc.
+    save_every_n_steps = 2160,
```
Comment (jiahe23) [inline examples/3dsphere/baroclinic_wave_rhoe.jl:19]:
```suggestion
    save_every_n_steps = 216,
```

File: examples/3dsphere/baroclinic_wave_rhotheta.jl [hunk 5]
```diff
@@ -12,8 +16,8 @@ driver_values(FT) = (;
     ode_algorithm = OrdinaryDiffEq.Rosenbrock23,
     jacobian_flags = (; ∂𝔼ₜ∂𝕄_mode = :exact, ∂𝕄ₜ∂ρ_mode = :exact),
     max_newton_iters = 2,
-    save_every_n_steps = 10,
-    additional_solver_kwargs = (;), # e.g., reltol, abstol, etc.
+    save_every_n_steps = 2160,
```
Comment (jiahe23) [inline examples/3dsphere/baroclinic_wave_rhotheta.jl:19]:
```suggestion
    save_every_n_steps = 216,
```

File: examples/3dsphere/driver.jl [hunk 6]
```diff
@@ -84,8 +107,9 @@ integrator = OrdinaryDiffEq.init(
     dt = dt,
     saveat = dt * save_every_n_steps,
     adaptive = false,
-    progress = true,
-    progress_steps = 1,
+    progress = false,
+    # progress = true,
+    # progress_steps = 1,
```
Comment (jiahe23) [inline examples/3dsphere/driver.jl:112]:
```suggestion
    progress = true,
    progress_steps = 1,
```

File: examples/3dsphere/held_suarez_rhoe.jl [hunk 7]
```diff
@@ -12,8 +16,8 @@ driver_values(FT) = (;
     ode_algorithm = OrdinaryDiffEq.Rosenbrock23,
     jacobian_flags = (; ∂𝔼ₜ∂𝕄_mode = :no_∂P∂K, ∂𝕄ₜ∂ρ_mode = :exact),
     max_newton_iters = 2,
-    save_every_n_steps = 10,
-    additional_solver_kwargs = (;), # e.g., reltol, abstol, etc.
+    save_every_n_steps = 2160,
```
Comment (jiahe23) [inline examples/3dsphere/held_suarez_rhoe.jl:19]:
```suggestion
    save_every_n_steps = 216,
```

File: examples/3dsphere/held_suarez_rhotheta.jl [hunk 8]
```diff
@@ -12,8 +16,8 @@ driver_values(FT) = (;
     ode_algorithm = OrdinaryDiffEq.Rosenbrock23,
     jacobian_flags = (; ∂𝔼ₜ∂𝕄_mode = :exact, ∂𝕄ₜ∂ρ_mode = :exact),
     max_newton_iters = 2,
-    save_every_n_steps = 10,
-    additional_solver_kwargs = (;), # e.g., reltol, abstol, etc.
+    save_every_n_steps = 2160,
```
Comment (jiahe23) [inline examples/3dsphere/held_suarez_rhotheta.jl:19]:
```suggestion
    save_every_n_steps = 216,
```

**Comment by jiahe23** [comment]
bors try

**Comment by jiahe23** [comment]
bors try-

**Comment by jiahe23** [comment]
bors try

**Pull request merged**